Mathlib/Data/Fintype/BigOperators.lean

```diff
@@ -295,4 +295,33 @@ theorem Fintype.prod_sum_type (f : Sum α₁ α₂ → M) :
 #align fintype.prod_sum_type Fintype.prod_sum_type
 #align fintype.sum_sum_type Fintype.sum_sum_type
 
+-- TODO: make sure this is inferred automatically from [Fintype α₁] and [Fintype α₂]
+variable [Fintype (α₁ × α₂)]
+
+@[to_additive Fintype.sum_prod_type]
+theorem Fintype.prod_prod_type [CommMonoid γ] {f : α₁ × α₂ → γ} :
+    ∏ x : α₁ × α₂, f x = ∏ x : α₁, ∏ y : α₂, f (x, y) :=
+  prod_finset_product univ univ (fun _ ↦ univ) <|
+      fun p => ⟨fun _ => ⟨mem_univ p.1, mem_univ p.2⟩, fun _ => mem_univ p⟩
+
+/-- An uncurried version of `Finset.prod_prod_type`. -/
+@[to_additive Fintype.sum_prod_type' "An uncurried version of `Finset.sum_prod_type`"]
+theorem Fintype.prod_prod_type' [CommMonoid γ] {f : α₁ → α₂ → γ} :
+    ∏ x : α₁ × α₂, f x.1 x.2 = ∏ x : α₁, ∏ y : α₂, f x y :=
+  prod_finset_product univ univ (fun _ ↦ univ) <|
+      fun p => ⟨fun _ => ⟨mem_univ p.1, mem_univ p.2⟩, fun _ => mem_univ p⟩
+
+@[to_additive Fintype.sum_prod_type_right]
+theorem Fintype.prod_prod_type_right [CommMonoid γ] {f : α₁ × α₂ → γ} :
+    ∏ x : α₁ × α₂, f x = ∏ y : α₂, ∏ x : α₁, f (x, y) :=
+  prod_finset_product_right univ univ (fun _ ↦ univ) <|
+      fun p => ⟨fun _ => ⟨mem_univ p.2, mem_univ p.1⟩, fun _ => mem_univ p⟩
```

Comment:
There should be a shorter proof via `prod_comm`.

Reply:
(which is perhaps an argument for skipping this lemma entirely)

Reply:
I propose we go either with what I have now, or remove the `_right` versions. Right now `prod_product_right` has one real usage and `prod_product_right'` has none.